p2p/test/transport/transport_test.go

```diff
@@ -220,12 +227,81 @@ func TestBigPing(t *testing.T) {
 	}
 }
 
+func TestManyBigPings(t *testing.T) {
+	// 64k buffer
+	const bufSize = 64 << 10
+	sendBuf := [bufSize]byte{}
+	const totalStreams = 512
+	const parallel = 8
+	// Total sends are > 20MiB
+	require.Greater(t, len(sendBuf)*totalStreams, 20<<20)
+	t.Log("Total sends:", len(sendBuf)*totalStreams)
+
+	// Fill with random bytes
+	_, err := rand.Read(sendBuf[:])
+	require.NoError(t, err)
+
+	for _, tc := range transportsToTest {
+		t.Run(tc.Name, func(t *testing.T) {
+			h1 := tc.HostGenerator(t, TransportTestCaseOpts{})
+			h2 := tc.HostGenerator(t, TransportTestCaseOpts{NoListen: true})
+			defer h1.Close()
+			defer h2.Close()
+			start := time.Now()
+			defer func() {
+				t.Log("Total time:", time.Since(start))
+			}()
+
+			require.NoError(t, h2.Connect(context.Background(), peer.AddrInfo{
+				ID:    h1.ID(),
+				Addrs: h1.Addrs(),
+			}))
+
+			h1.SetStreamHandler("/BIG-ping/1.0.0", func(s network.Stream) {
```

Comment:
```suggestion
			h1.SetStreamHandler("/big-ping", func(s network.Stream) {
```

Reply:
It's capital because it's BIG! I'll change though